### ISSUE TITLE
Podman does not work with users created by homectl/systemd-homed.

### ISSUE THREAD

**Issue by ananthb** [containers/podman#12590]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

All podman root-less commands fail with `Error: cannot setup namespace using newuidmap: exit status 1` with a user managed by systemd-homed.

**Steps to reproduce the issue:**

1. Activate systemd-homed service, and create a user account with homectl.
2. Login as the newly created user and run any podman command.

**Describe the results you received:**

Podman fails to run.

**Describe the results you expected:**

Podman should run.

**Additional information you deem important (e.g. issue happens only occasionally):**

The user's home is managed by [systemd-homed](https://systemd.io/HOME_DIRECTORY/). No user information is stored in /etc/passwd, /etc/shadow and /etc/gshadow.
User records are created on the fly by systemd-homed.

**Output of `podman version`:**

```
podman version 3.4.4
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: /usr/bin/conmon is owned by conmon 1:2.0.31-1
    path: /usr/bin/conmon
    version: 'conmon version 2.0.31, commit: 7e7eb74e52abf65a6d46807eeaea75425cc8a36c'
  cpus: 16
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: shadowfax
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 60237
      size: 1
    uidmap:
    - container_id: 0
      host_id: 60237
      size: 1
  kernel: 5.15.7-arch1-1
  linkmode: dynamic
  logDriver: journald
  memFree: 252006400
  memTotal: 33668792320
  ociRuntime:
    name: crun
    package: /usr/bin/crun is owned by crun 1.3-3
    path: /usr/bin/crun
    version: |-
      crun version 1.3
      commit: 4f6c8e0583c679bfee6a899c05ac6b916022561b
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/60237/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: /usr/bin/slirp4netns is owned by slirp4netns 1.1.12-1
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 21445206016
  swapTotal: 21474828288
  uptime: 28h 37m 3.96s (Approximately 1.17 days)
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/ananth/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/ananth/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: f2fs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/user/60237/containers
  volumePath: /home/ananth/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.4
  Built: 1639074640
  BuiltTime: Fri Dec 10 00:00:40 2021
  GitCommit: f6526ada1025c2e3f88745ba83b8b461ca659933
  GoVersion: go1.17.4
  OsArch: linux/amd64
  Version: 3.4.4
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
Name            : podman
Version         : 3.4.4-1
Description     : Tool and library for running OCI-based containers in pods
Architecture    : x86_64
URL             : https://github.com/containers/podman
Licenses        : Apache
Groups          : None
Provides        : None
Depends On      : cni-plugins  conmon  containers-common  crun  fuse-overlayfs  iptables  libdevmapper.so=1.02-64  libgpgme.so=11-64  libseccomp.so=2-64  slirp4netns
Optional Deps   : apparmor: for AppArmor support [installed]
                  btrfs-progs: support btrfs backend devices [installed]
                  catatonit: --init flag support [installed]
                  podman-docker: for Docker-compatible CLI
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 72.79 MiB
Packager        : David Runge <dvzrv@archlinux.org>
Build Date      : Friday 10 December 2021 12:00:40 AM
Install Date    : Friday 10 December 2021 05:02:42 PM
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Arch Linux running on a physical computer.

**Comment by rhatdan** [comment]
Looks like no /etc/subuid and /etc/subgid is being configured, which is pretty much required for rootless podman.

**Comment by ananthb** [comment]
I ran into this error only with /etc/subuid and /etc/subgid configured. I deleted them just before opening this card to see what would happen if I ran without it. It did run but it printed a warning about missing subuids first.

**Comment by rhatdan** [comment]
If the /etc/subuid and /etc/subgid accounts are back in place, can you do

$ cat /etc/subuid
$ cat /etc/subgid
$ podman system migrate
$ podman unshare cat /proc/self/uid_map

**Comment by rhatdan** [comment]
@giuseppe FYI

**Comment by ananthb** [comment]
I've got a disk image of the system that caused this issue. Had to reinstall to have a usable system again. I suspect that systemd-homed's fscrypt implementation may have something to do with this error. I've switched to the LUKS backend, and podman user mode runs absolutely file.

Here's a bug caused by systemd-homed's implementation of fscrypt that breaks docker: https://github.com/google/fscrypt/issues/128. I suspect something similar may be happening with podman. Though not this exact issue because of architecture differences between docker and podman.

I'll recreate this error on my saved disk image and report my findings.

**Comment by ananthb** [comment]
/etc/subuid
`ananth:100000:65536
`
/etc/subgid
`ananth:100000:65536
`
podman system migrate
`Error: cannot setup namespace using newuidmap: exit status 1
`
podman unshare cat /proc/self/uid_map
`Error: cannot setup namespace using newuidmap: exit status 1`

@rhatdan

**Comment by rhatdan** [comment]
Ok, one other thing would be if the process has already dropped privs. One thought would be you are already in a user namespace or a second one would be you already dropped caps.

$  cat /proc/sef/uid_map
$ grep Cap /proc/self/status
CapInh:	0000000000000000
CapPrm:	0000000000000000
CapEff:	0000000000000000
CapBnd:	000001ffffffffff
CapAmb:	0000000000000000

**Comment by giuseppe** [comment]
another thing to check is the mount table.  It could be systemd to mount `/usr` with `nosuid`

**Comment by giuseppe** [comment]
could you share the output of `cat /proc/self/mountinfo`?

**Comment by rhatdan** [comment]
Similarly could you check if sudo works in this environment?
$ grep -i new /proc/self/status 
NoNewPrivs:	0

**Comment by ananthb** [comment]
```
$ cat /proc/self/uid_map
         0          0 4294967295

$ cat /proc/self/mountinfo
24 63 0:22 / /proc rw,nosuid,nodev,noexec,relatime shared:12 - proc proc rw
25 63 0:23 / /sys rw,nosuid,nodev,noexec,relatime shared:2 - sysfs sysfs rw
26 63 0:5 / /dev rw,nosuid shared:8 - devtmpfs devtmpfs rw,size=16396092k,nr_inodes=4099023,mode=755,inode64
27 25 0:6 / /sys/kernel/security rw,nosuid,nodev,noexec,relatime shared:3 - securityfs securityfs rw
28 26 0:24 / /dev/shm rw,nosuid,nodev shared:9 - tmpfs tmpfs rw,inode64
29 26 0:25 / /dev/pts rw,nosuid,noexec,relatime shared:10 - devpts devpts rw,gid=5,mode=620,ptmxmode=000
30 63 0:26 / /run rw,nosuid,nodev shared:11 - tmpfs tmpfs rw,size=6575936k,nr_inodes=819200,mode=755,inode64
31 25 0:27 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime shared:4 - cgroup2 cgroup2 rw,nsdelegate,memory_recursiveprot
32 25 0:28 / /sys/fs/pstore rw,nosuid,nodev,noexec,relatime shared:5 - pstore pstore rw
33 25 0:29 / /sys/firmware/efi/efivars rw,nosuid,nodev,noexec,relatime shared:6 - efivarfs efivarfs rw
34 25 0:30 / /sys/fs/bpf rw,nosuid,nodev,noexec,relatime shared:7 - bpf none rw,mode=700
63 1 254:0 / / rw,relatime shared:1 - f2fs /dev/mapper/root0 rw,lazytime,background_gc=on,discard,no_heap,user_xattr,inline_xattr,acl,inline_data,inline_dentry,flush_merge,extent_cache,mode=adaptive,active_logs=6,alloc_mode=default,checkpoint_merge,fsync_mode=posix,compress_algorithm=zstd,compress_log_size=2,compress_mode=fs,discard_unit=block
35 24 0:31 / /proc/sys/fs/binfmt_misc rw,relatime shared:13 - autofs systemd-1 rw,fd=31,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=26629
36 26 0:32 / /dev/hugepages rw,relatime shared:14 - hugetlbfs hugetlbfs rw,pagesize=2M
37 26 0:21 / /dev/mqueue rw,nosuid,nodev,noexec,relatime shared:15 - mqueue mqueue rw
38 25 0:7 / /sys/kernel/debug rw,nosuid,nodev,noexec,relatime shared:16 - debugfs debugfs rw
39 25 0:12 / /sys/kernel/tracing rw,nosuid,nodev,noexec,relatime shared:17 - tracefs tracefs rw
40 35 0:33 / /proc/sys/fs/binfmt_misc rw,nosuid,nodev,noexec,relatime shared:18 - binfmt_misc binfmt_misc rw
41 25 0:34 / /sys/kernel/config rw,nosuid,nodev,noexec,relatime shared:19 - configfs configfs rw
43 25 0:35 / /sys/fs/fuse/connections rw,nosuid,nodev,noexec,relatime shared:20 - fusectl fusectl rw
137 63 259:1 / /boot rw,relatime shared:64 - vfat /dev/nvme0n1p1 rw,lazytime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro
142 63 0:38 / /tmp rw,nosuid,nodev shared:72 - tmpfs tmpfs rw,nr_inodes=409600,inode64
416 63 0:52 /ananth /home/ananth rw,nosuid,nodev,relatime shared:89 - btrfs /dev/mapper/home-ananth rw,ssd,discard,noacl,space_cache=v2,subvolid=256,subvol=/ananth
396 30 0:45 / /run/user/60237 rw,nosuid,nodev,relatime shared:248 - tmpfs tmpfs rw,size=3287968k,nr_inodes=821992,mode=700,uid=60237,gid=60237,inode64
457 396 0:47 / /run/user/60237/gvfs rw,nosuid,nodev,relatime shared:260 - fuse.gvfsd-fuse gvfsd-fuse rw,user_id=60237,group_id=60237
400 396 0:59 / /run/user/60237/doc rw,nosuid,nodev,relatime shared:205 - fuse.portal portal rw,user_id=60237,group_id=60237

$ grep -i new /proc/self/status
NoNewPrivs:	0
```

@giuseppe @rhatdan

**Comment by PavelSosin-320** [comment]
In Fedora 35 Workstation:
/proc/mounts
[pavelsosin@Dell ~]$ cat /proc/mounts
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
sysfs /sys sysfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0
devtmpfs /dev devtmpfs rw,seclabel,nosuid,size=3971500k,nr_inodes=992875,mode=755,inode64 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev/shm tmpfs rw,seclabel,nosuid,nodev,inode64 0 0
devpts /dev/pts devpts rw,seclabel,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
tmpfs /run tmpfs rw,seclabel,nosuid,nodev,size=1597172k,nr_inodes=819200,mode=755,inode64 0 0
cgroup2 /sys/fs/cgroup cgroup2 rw,seclabel,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot 0 0
pstore /sys/fs/pstore pstore rw,seclabel,nosuid,nodev,noexec,relatime 0 0
efivarfs /sys/firmware/efi/efivars efivarfs rw,nosuid,nodev,noexec,relatime 0 0
none /sys/fs/bpf bpf rw,nosuid,nodev,noexec,relatime,mode=700 0 0
/dev/sda3 / btrfs rw,seclabel,relatime,compress=zstd:1,space_cache,subvolid=258,subvol=/root 0 0
selinuxfs /sys/fs/selinux selinuxfs rw,nosuid,noexec,relatime 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=31,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=16663 0 0
mqueue /dev/mqueue mqueue rw,seclabel,nosuid,nodev,noexec,relatime 0 0
tracefs /sys/kernel/tracing tracefs rw,seclabel,nosuid,nodev,noexec,relatime 0 0
configfs /sys/kernel/config configfs rw,nosuid,nodev,noexec,relatime 0 0
debugfs /sys/kernel/debug debugfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0
fusectl /sys/fs/fuse/connections fusectl rw,nosuid,nodev,noexec,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,seclabel,relatime,pagesize=2M 0 0
tmpfs /tmp tmpfs rw,seclabel,nosuid,nodev,size=3992932k,nr_inodes=409600,inode64 0 0
/dev/sda3 /home btrfs rw,seclabel,nosuid,nodev,noexec,relatime,compress=zstd:1,space_cache,subvolid=256,subvol=/home 0 0
/dev/sda2 /boot ext4 rw,seclabel,relatime 0 0
/dev/sda1 /boot/efi vfat rw,relatime,fmask=0077,dmask=0077,codepage=437,iocharset=ascii,shortname=winnt,errors=remount-ro 0 0
sunrpc /var/lib/nfs/rpc_pipefs rpc_pipefs rw,relatime 0 0
tmpfs /run/user/1001 tmpfs rw,seclabel,nosuid,nodev,relatime,size=798584k,nr_inodes=199646,mode=700,uid=1001,gid=1001,inode64 0 0
tmpfs /run/user/1000 tmpfs rw,seclabel,nosuid,nodev,relatime,size=798584k,nr_inodes=199646,mode=700,uid=1000,gid=1000,inode64 0 0
gvfsd-fuse /run/user/1001/gvfs fuse.gvfsd-fuse rw,nosuid,nodev,relatime,user_id=1001,group_id=1001 0 0
gvfsd-fuse /run/user/1000/gvfs fuse.gvfsd-fuse rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
portal /run/user/1000/doc fuse.portal rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
portal /run/user/1001/doc fuse.portal rw,nosuid,nodev,relatime,user_id=1001,group_id=1001 0 0
gvfsd-fuse /root/.cache/gvfs fuse.gvfsd-fuse rw,nosuid,nodev,relatime,user_id=0,group_id=0 0 0

Root user in the userdb is marked as "nologin" - it doesn't start GNOME session and can't be logged-in via httpd!
It works like Windows where no root user exists and Admin is wheel-like.
After sudo  
podman create works ! But btrfs subvolume get-default shows /home && echo $HOME shows /root. Then everything is created under the /root/containers/ --> root subvolume. Root is homeless.

**Comment by PavelSosin-320** [comment]
There is home.mount unit exists in the GNOME user's graph. Maybe, its status check is necessary to ensure that Podman has storage at all when used by roootless users?

**Comment by fansari** [comment]
I have a simular  issue. I want to test systemd-homed and create rootless containers. The issues start when I try to pull an image.

```
[fansari@bat ldap]$ docker pull alpine                                                                                                                                                                                               
Resolved "alpine" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)                                                                                                                                                         
Trying to pull docker.io/library/alpine:latest...                                                                                                                                                                                             
Getting image source signatures                                                                                                                                                                                                               
Copying blob 59bf1c3509f3 done                                                                                                                                                                                                                
Error: writing blob: adding layer with blob "sha256:59bf1c3509f33515622619af21ed55bbe26d24913cedbca106468a5fb37a50c3": Error processing tar file(exit status 1): potentially insufficient UIDs or GIDs available in user namespace (requested 
0:42 for /etc/shadow): Check /etc/subuid and /etc/subgid: lchown /etc/shadow: invalid argument
```

Tested with Fedora 35 Sliverblue.

```
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.30-2.fc35.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.30, commit: '
  cpus: 8
  distribution:
    distribution: fedora
    variant: silverblue
    version: "35"
  eventLogger: journald
  hostname: bat.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 60257
      size: 1
    uidmap:
    - container_id: 0
      host_id: 60257
      size: 1
  kernel: 5.15.10-200.fc35.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 3916976128
  memTotal: 8205803520
  ociRuntime:
    name: crun
    package: crun-1.3-1.fc35.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.3
      commit: 8e5757a4e68590326dafe8a8b1b4a584b10a1370
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/60257/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc35.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 16795033600
  swapTotal: 16795033600
  uptime: 14m 56.03s
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /var/home/fansari/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.7.1-2.fc35.x86_64
      Version: |-
        fusermount3 version: 3.10.5
        fuse-overlayfs: version 1.7.1
        FUSE library version 3.10.5
        using FUSE kernel interface version 7.31
  graphRoot: /var/mnt/data/podman/fansari/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/user/60257
  volumePath: /var/mnt/data/podman/fansari/containers/storage/volumes
version:
  APIVersion: 3.4.4
  Built: 1638999907
  BuiltTime: Wed Dec  8 22:45:07 2021
  GitCommit: ""
  GoVersion: go1.16.8
  OsArch: linux/amd64
  Version: 3.4.4
```

**Comment by PavelSosin-320** [comment]
@fansari I tried almost every combination: btrfs driver + fuse-mount, default mount, wtc. Nothing works. The message that I get references STAT  error. According to [Relese notes 4.13](https://btrfs.wiki.kernel.org/index.php/Changelog#btrfs-progs_v5.15_.28Nov_2021.29) <html>
<body>


overlayfs | 4.15 | Overlayfs can now use btrfs as the lower filesystem.
-- | -- | --



</body>
</html>
But it still doesn't work with our btrfs 5.15 or Podman is built with wrong library. Indeed, I didn't succeed to build Containers-storage CLI in my Fedora 35 WS. 
Neither log file related to SELinux contains something related to the STAT Is Container-storage too old?

**Comment by fansari** [comment]
I have created a new user with homectl with  fs-type=xfs. The error messages look the same.

```
Resolved "alpine" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)                                                                                                                                                         
Trying to pull docker.io/library/alpine:latest...                                                                                                                                                                                             
Getting image source signatures                                                                                                                                                                                                               
Copying blob sha256:59bf1c3509f33515622619af21ed55bbe26d24913cedbca106468a5fb37a50c3                                                                                                                                                          
Copying blob sha256:59bf1c3509f33515622619af21ed55bbe26d24913cedbca106468a5fb37a50c3                                                                                                                                                          
Error: writing blob: adding layer with blob "sha256:59bf1c3509f33515622619af21ed55bbe26d24913cedbca106468a5fb37a50c3": Error processing tar file(exit status 1): potentially insufficient UIDs or GIDs available in user namespace (requested 
0:42 for /etc/shadow): Check /etc/subuid and /etc/subgid: lchown /etc/shadow: invalid argument      
```

**Comment by fansari** [comment]
I found the issue. I followed this guide:

[https://unix.stackexchange.com/questions/176593/how-to-influence-the-assignment-of-subordinate-uids-gids-when-creating-user-acco](url)

```
user=<insertusername>
uid=$(id -u $user)
sudo usermod -v $((uid+200000))-$((uid+200000+65535)) -w $((uid+200000))-$((uid+200000+65535)) $user
```
This created the missing entries in /etc/subuid and /etc/subgid.

Seems that when you don't use systemd-homed all this is done automatically.

After setting this up I was able to pull and image.

**Comment by mybigman** [comment]
@fansari still doesnt resolve this issue for me I get the same error as you https://github.com/containers/podman/issues/12590#issuecomment-1003027364

```shell
> cat subgid
champ:260083:65536

> cat subuid
champ:260083:65536
```

**Comment by rhatdan** [comment]
What does `podman unshare cat /proc/self/uid_map` show?
Can you execute `podman system migrate` and try this again.

**Comment by ananthb** [comment]
I re-created the home directory using LUKS storage and the error is gone. It could be related to systemd-homed using fscrypt v1 encryption policy.

**Comment by mybigman** [comment]
@rhatdan 

Some additional info.

ArchLinux Arm, luks with btrfs FS, 2 x sub volumes @ and @home, using systemd-homed.

```
cat /etc/sub*
champ:260083:1694527157
champ:260083:1694527157
```

```
podman unshare cat /proc/self/uid_map
         0      60083          1
         1     260083 1694527157
```

`podman system migrate` yields no error

```
podman pull docker.io/busybox
Trying to pull docker.io/library/busybox:latest...
Getting image source signatures
Copying blob a01966dde7f8 done
Error: writing blob: adding layer with blob "sha256:a01966dde7f8d5ba10b6d87e776c7c8fb5a5f6bfa678874bd28b33b1fc6dba34": Error processing tar file(exit status 1): potentially insufficient UIDs or GIDs available in user namespace (requested 65534:65534 for /home): Check /etc/subuid and /etc/subgid: lchown /home: invalid argument
```

```
podman info
host:
  arch: arm64
  buildahVersion: 1.23.1
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: /usr/bin/conmon is owned by conmon 1:2.0.31-1
    path: /usr/bin/conmon
    version: 'conmon version 2.0.31, commit: 7e7eb74e52abf65a6d46807eeaea75425cc8a36c'
  cpus: 2
  distribution:
    distribution: archarm
    version: unknown
  eventLogger: journald
  hostname: arch
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 60083
      size: 1
    - container_id: 1
      host_id: 260083
      size: 1694527157
    uidmap:
    - container_id: 0
      host_id: 60083
      size: 1
    - container_id: 1
      host_id: 260083
      size: 1694527157
  kernel: 5.15.5-1-ARCH
  linkmode: dynamic
  logDriver: journald
  memFree: 2784391168
  memTotal: 3118075904
  ociRuntime:
    name: crun
    package: /usr/bin/crun is owned by crun 1.4-1
    path: /usr/bin/crun
    version: |-
      crun version 1.4
      commit: 3daded072ef008ef0840e8eccb0b52a7efbd165d
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/60083/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: /usr/bin/slirp4netns is owned by slirp4netns 1.1.12-1
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 1558704128
  swapTotal: 1558704128
  uptime: 17m 31.48s
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries: {}
store:
  configFile: /home/champ/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: /usr/bin/fuse-overlayfs is owned by fuse-overlayfs 1.8-1
      Version: |-
        fusermount3 version: 3.10.5
        fuse-overlayfs: version 1.8
        FUSE library version 3.10.5
        using FUSE kernel interface version 7.31
  graphRoot: /home/champ/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/user/60083/containers
  volumePath: /home/champ/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.4
  Built: 1639077677
  BuiltTime: Fri Dec 10 06:21:17 2021
  GitCommit: f6526ada1025c2e3f88745ba83b8b461ca659933
  GoVersion: go1.17.4
  OsArch: linux/arm64
  Version: 3.4.4
```

**Comment by PavelSosin-320** [comment]
@fansari Why error message blames STAT system call and not create ? Did create succeed ? I looked inside the container-storage source and didn't find single line of code that communicates with systemd-homed service or checks home area status. Homed unit status is OK but I don't know if it is enough to assume that home area is useful in the current state of GUI session started by systemd-user. Rootless container that requires uid_map and regular container artifact creation are 2 different scenario but if containers-storage ignores home area the result will be the same. I afraid that homed unit execution is also incomplete.

**Comment by fansari** [comment]
I use Fedora 35 Silverblue but as I found with this issue and others the distribution is still not really ready for systemd-homed.
One of the issues is that these subuids and subguids are not created automatically for podman. Other issues are missing SELinux policy so you can only use it in permissive mode. Also gdm is not aware of the user so you have to create a user with "normal" home dir before you can use a user for systemd-homed. This is a little bit "offtopic" of course but it shows that the issue with podman is not the only issue where systemd-homed is not integrated well in current Linux distributions.

**Comment by rhatdan** [comment]
What SELinux AVCs are you seeing?

Have you opened issues with systemd-homed?  It should work similar to useradd when configuring the /etc/subuid and /etc/subgid files.  Or work on a newer solution to allow user namespace support the generated user.

**Comment by fansari** [comment]
I have created a logfile and added it to my bug report.

https://bugzilla.redhat.com/show_bug.cgi?id=2036108

**Comment by giuseppe** [comment]
could https://github.com/systemd/systemd/issues/21952 be the cause of this issue?

**Comment by PavelSosin-320** [comment]
The [Christopher Smart post also mentions this issue](https://blog.christophersmart.com/2021/01/31/podman-volumes-and-selinux/) 1 year ago. 
I see another issue: btrfs subvolume create works but "subvolume directory" unlabeled at all unlike any other directory in the home area. What getsubids -g pavelsosin
0: pavelsosin 100000 65536 does **100000** mean, user range starting from 100 thousands? Is it "nobody"? According to
       _Subuid authorizes a user id to map ranges of user ids from its
       namespace into child namespaces._ 
How SELinux handles nobody?

**Comment by rhatdan** [comment]
SELinux does not care about users.  Only about XAttrs on files.  If a subvolume directory is unlabled, and this gets used for container storage, then SELinux will block containers from using it.

What getsubids -g pavelsosin
0: pavelsosin 100000 65536 
I believe this means that pavelsosin has access to UIDs 100000 and then next 65534 UIDs and can assign it to a user namespace.

**Comment by PavelSosin-320** [comment]
@rhatdan I believe that it is correct: the issue is that btrfs subvolume create command and/or library function are disconnected from SELinux and when containers storage creates container as btrfs  subvolume it has to care about labelling of subvolume "directory" and maybe, its content too. In other words, "create" works but creates something not usable due to SELinux restrictions. Another scenario can be that the SELinux library function is called but the issue with this function is hidden in the SELinux issues list.

**Comment by PavelSosin-320** [comment]
Perfect! This issue is on and off starting from Fedora 26 [RedHat bugzilla err 1447750](https://bugzilla.redhat.com/show_bug.cgi?id=1447750), has duplicates in some products and closed due to Fedora 26 EOL without solution.

**Comment by rhatdan** [comment]
The last comment in the bugzilla states that it was fixed in the f27 kernel.

@wrabcak  can you get someone from SELinux to look at this?

**Comment by wrabcak** [comment]
Sure, 
@omos @plautrba @vmojzis @zpytela PTAL.

**Comment by PavelSosin-320** [comment]
In other words, pavel sosin can access  65354 "nobodies" because id's starting from 100000 belongs to the "nobody" diapazone. I'm not sure that "nobodies" are not restricted by something, like polkitt. It also assumes that user/group ids are long" and used as long in every API invocation. Users created by Anakonda have 16bits length ids. Long UID/GIDs may affect [User namespaces](https://man7.org/linux/man-pages/man7/user_namespaces.7.html). Anakonda itself is not implemented perfectly in Fedora 35.

**Comment by anthr76** [comment]
https://github.com/systemd/systemd/pull/22239

Looks like this will unblock this

**Comment by ananthb** [comment]
Yeah this will fix homectl users. Users created with --storage=fscrypt are still blocked by systemd-homed using v1 encryption policies but that's also being discussed upstream. Btrfs and subvolume based homes should start working if the uid mapping is fixed.

**Comment by anthr76** [comment]
I'm testing the above patch [here](https://aur.archlinux.org/packages/systemd-homed-containers/) and it's working well with a btrfs subvol

**Comment by giuseppe** [comment]
> I'm testing the above patch [here](https://aur.archlinux.org/packages/systemd-homed-containers/) and it's working well with a btrfs subvol

Thanks for confirming it!

I think we can close this issue since there is nothing to do on the Podman side.
